File: K_Radius_Subarray_Averages.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var getAverages = function(nums, k) {
    const n = nums.length;
    const avgs = new Array(n).fill(-1);
    const target_length = k * 2 + 1;
    let total = nums.slice(0, target_length).reduce((acc, val) => acc + val, 0);

    for (let c = k; c < n; c++) {
        if (c + k < n) {
            avgs[c] = Math.floor(total / target_length);
            if (c + k + 1 >= n) {
                break;
            }
            total -= nums[c - k];
            total += nums[c + k + 1];
        }
    }

    return avgs;    
};

File: K_Radius_Subarray_Averages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        avgs = [-1] * n
        target_length = k * 2 + 1
        total = sum(nums[0:target_length])

        for c in range(k, n):
            if c + k < n:
                avgs[c] = total // target_length
                if c + k + 1 >= n:
                    break
                total -= nums[c - k]
                total += nums[c + k + 1]
        
        return avgs
